cdk-infra/lib/bulkExportStateMachine.ts

```diff
@@ -0,0 +1,91 @@
+import { LambdaInvoke } from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+
+export default class BulkExportStateMachine {
+    bulkExportStateMachine: StateMachine;
+
+    constructor(
+        scope: Construct,
+        updateStatusLambdaFunction: Function,
+        startExportJobLambdaFunction: Function,
+        getExportJobLambdaFunction: Function,
+        stopExportJobLambdaFunction: Function,
+    ) {
+        const catchAllUpdateStatusToFailed = new LambdaInvoke(scope, 'catchAllUpdateStatusToFailed', {
+            lambdaFunction: updateStatusLambdaFunction,
+            payload: TaskInput.fromObject({
+                'globalParams.$': '$',
+                status: 'failed',
+            }),
+        });
+
+        const updateStatusToFailed = new LambdaInvoke(scope, 'updateStatusToFailed', {
+            lambdaFunction: updateStatusLambdaFunction,
+            payload: TaskInput.fromObject({
+                'globalParams.$': '$',
+                status: 'failed',
+            }),
+        });
+
+        const updateStatusToCanceled = new LambdaInvoke(scope, 'updateStatusToCanceled', {
+            lambdaFunction: updateStatusLambdaFunction,
+            payload: TaskInput.fromObject({
+                'globalParams.$': '$',
+                status: 'canceled',
+            }),
+        });
+
+        const updateStatusToCompleted = new LambdaInvoke(scope, 'updateStatusToCompleted', {
+            lambdaFunction: updateStatusLambdaFunction,
+            payload: TaskInput.fromObject({
+                'globalParams.$': '$',
+                status: 'completed',
+            }),
+        });
+
+        const stopExportJob = new LambdaInvoke(scope, 'stopExportJob', {
+            lambdaFunction: stopExportJobLambdaFunction,
+        }).next(updateStatusToCanceled);
+
+        const waitForExportJob = new Wait(scope, 'waitForExportJob', {
+            time: WaitTime.duration(Duration.seconds(10)),
+        });
+
+        const choiceOnJobStatus = new Choice(scope, 'choiceOnJobStatus')
+            .when(Condition.booleanEquals('$.executionParameters.isCanceled', true), stopExportJob)
+            .when(
+                Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'SUCCEEDED'),
+                updateStatusToCompleted,
+            )
+            .when(Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'STARTING'), waitForExportJob)
+            .when(Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'RUNNING'), waitForExportJob)
+            .when(Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'FAILED'), updateStatusToFailed)
+            .when(Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'TIMEOUT'), updateStatusToFailed)
+            .when(Condition.stringEquals('$.executionParameters.glueJobRunStatus', 'STOPPING'), updateStatusToFailed)
```

Comment:
Should 'STOPPING' and 'STOPPED' map to 'updateStatusToCanceled'?

Reply:
I've added a comment explaining why we don't map to the canceled state, but in essence STOPPING and STOPPED can only occur if the job was forcefully stopped with a Glue API call from outside the FHIR server, so we treat it as failure